src/Hydra/Serializer/PartialCollectionViewNormalizer.php

```diff
@@ -71,12 +75,40 @@ public function normalize($object, $format = null, array $context = [])
             return $data;
         }
 
+        $metadata = isset($context['resource_class']) ? $this->resourceMetadataFactory->create($context['resource_class']) : null;
+        $isPaginatedWithCursor = $paginated && null !== $metadata && null !== $cursorPaginationAttribute = $metadata->getAttribute('pagination_via_cursor');
+
         $data['hydra:view'] = [
-            '@id' => IriHelper::createIri($parsed['parts'], $parsed['parameters'], $this->pageParameterName, $paginated ? $currentPage : null),
+            '@id' => IriHelper::createIri($parsed['parts'], $parsed['parameters'], $this->pageParameterName, $paginated && !$isPaginatedWithCursor ? $currentPage : null),
             '@type' => 'hydra:PartialCollectionView',
         ];
 
-        if ($paginated) {
+        if ($isPaginatedWithCursor) {
+            $forwardRangeOperator = 'desc' === strtolower($cursorPaginationAttribute['direction']) ? 'lt' : 'gt';
+            $backwardRangeOperator = 'gt' === $forwardRangeOperator ? 'lt' : 'gt';
+
+            $accessor = PropertyAccess::createPropertyAccessor();
```

Comment:
You should reuse the existing service if it exists (and create it only if `null` is passed in the constructor). This way we'll benefit from the component's cache.